### PR TITLE
PHOENIX-5209 Cannot add non-PK column to table when the last PK column is of type VARBINARY or ARRAY

### PR DESCRIPTION
**Pull request by brfrn169**

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java [hunk 1]
```diff
@@ -3593,10 +3593,11 @@ public MutationState addColumn(PTable table, List<ColumnDef> origColumnDefs,
...
                 int position = table.getColumns().size();
 
-                List<PColumn> currentPKs = table.getPKColumns();
-                if (numCols > 0) {
+                boolean addPKColumns = columnDefs.stream().anyMatch(ColumnDef::isPK);
+                if (addPKColumns) {
+                    List<PColumn> currentPKs = table.getPKColumns();
                     PColumn lastPK = currentPKs.get(currentPKs.size()-1);
-                    // Disallow adding columns if the last column is VARBIANRY.
+                    // Disallow adding columns if the last column is VARBIANRY or ARRAY.
```
Comment (swaroopak) [inline phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java]: (unrelated to this PR) nit: last column in the primary key.

File: phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java [hunk 2]
```diff
@@ -3593,10 +3593,11 @@ public MutationState addColumn(PTable table, List<ColumnDef> origColumnDefs,
 
                 int position = table.getColumns().size();
 
-                List<PColumn> currentPKs = table.getPKColumns();
-                if (numCols > 0) {
+                boolean addPKColumns = columnDefs.stream().anyMatch(ColumnDef::isPK);
```
Comment (swaroopak) [inline phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java]: nice one-liner change!

Comment (brfrn169) [reply phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java]: Do you mean I should remove the `addPKColumns` variable and change the if statement to something like the following:
```
                if (columnDefs.stream().anyMatch(ColumnDef::isPK)) {
```

Comment (swaroopak) [reply phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java]: I wasn't suggesting anything. I rather appreciated your change :)

Comment (brfrn169) [reply phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java]: Sure. Thank you for reviewing. I will change the patch for your review.

**Comment by brfrn169** [comment]
I added a new commit for your review. Can you please review it? @swaroopak

**Comment by gjacoby126** [comment]
@swaroopak - I see you approved this a few days back -- if you think it's ready to go, could you please commit to the appropriate branches? Thanks!

**Pull request merged**